Parse unlinkat flags

arossert opened the pull request:
This add parsing to `unlinkat` flags argument as requested in #803

mstemm commented on driver/event_table.c:
```diff
@@ -308,4 +308,6 @@ const struct ppm_event_info g_event_info[PPM_EVENT_MAX] = {
 	/* PPME_SYSCALL_EXECVE_17_X */{"execve", EC_PROCESS, EF_MODIFIES_STATE, 17, {{"res", PT_ERRNO, PF_DEC}, {"exe", PT_CHARBUF, PF_NA}, {"args", PT_BYTEBUF, PF_NA}, {"tid", PT_PID, PF_DEC}, {"pid", PT_PID, PF_DEC}, {"ptid", PT_PID, PF_DEC}, {"cwd", PT_CHARBUF, PF_NA}, {"fdlimit", PT_UINT64, PF_DEC}, {"pgft_maj", PT_UINT64, PF_DEC}, {"pgft_min", PT_UINT64, PF_DEC}, {"vm_size", PT_UINT32, PF_DEC}, {"v ...
 	/* PPME_SYSCALL_UNSHARE_E */ {"unshare", EC_PROCESS, EF_NONE, 1, {{"flags", PT_FLAGS32, PF_HEX, clone_flags} } },
 	/* PPME_SYSCALL_UNSHARE_X */ {"unshare", EC_PROCESS, EF_NONE, 1, {{"res", PT_ERRNO, PF_DEC} } },
+	/* PPME_SYSCALL_UNLINKAT_3_E */{"unlinkat", EC_FILE, EF_NONE, 3, {{"dirfd", PT_FD, PF_DEC}, {"name", PT_CHARBUF, PF_NA}, {"flags", PT_FLAGS8, PF_HEX, unlinkat_flags} } },
```
Why the 3 here, and not UNLINKAT_2_{E,X}?

arossert replied:
As far as I understand the "version" represent the number of parameters that the syscall represent.
Take `execve` for example, the first "version" is `PPME_SYSCALL_EXECVE_8_{E,X}` that handles 8 parameters, then the next version is `PPME_SYSCALL_EXECVE_13_{E,X}` that handles 13 parameters... and so on.
If I misunderstood convention this I can change it to `UNLINKAT_2_{E,X}`.

luca3m replied:
3 is just an incremental version number, it does not represent argument count

arossert replied:
Changed to PPME_SYSCALL_UNLINKAT_2_{E/X}

mstemm commented on driver/ppm_fillers.c:
```diff
@@ -243,8 +244,8 @@ const struct ppm_event_entry g_ppm_events[PPM_EVENT_MAX] = {
 	[PPME_SYSCALL_UNLINK_X] = {f_sys_single_x},
-	[PPME_SYSCALL_UNLINKAT_E] = {PPM_AUTOFILL, 2, APT_REG, {{0}, {1} } },
-	[PPME_SYSCALL_UNLINKAT_X] = {f_sys_single_x},
+	[PPME_SYSCALL_UNLINKAT_3_E] = {f_sys_unlinkat_e},
```
Remind me why PPM_AUTOFILL doesn't work here?

arossert replied:
Can `PPM_AUTOFILL` handle syscall flags?
I have a call to convert the flags to `PPM_` flags:
`static inline u32 unlinkat_flags_to_scap(unsigned long flags)`
(From the [documentation](https://linux.die.net/man/2/unlinkat) there is only one flag supported `AT_REMOVEDIR`)

mstemm commented:
Did you want filterchecks to tests these flags as well? Otherwise not sure how you'll use them.

arossert commented:
@mstemm I have created this PR following #803.
I'm not sure how to incorporate this in filterchecks, will this be part of existing `filtercheck_field_info` or a new one need to be created for this?
With some guidance, I can try doing this.

mstemm commented:
Ah, if you just want them displayed as a part of evt.args for an unlinkat event, these changes should be fine.

arossert commented:
I have added a fix following the latest #1025.
@mstemm, I have moved the `name` argument to the exit event (as userspace pointers should be), should I do the same for `unlink` syscall in this PR?

mattpag commented:
Closing, subset of https://github.com/draios/sysdig/pull/1076
